### PR TITLE
windows_compatibility_tari_base_node

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- Conditional compilation was added to exclude `store_forward` Tokio layer to be added for Windows builds. This removes  `store_forward` functionality from Windpows, but provides a temporary fix.
- Small refactoring for `fn load_users()` in `lmdb.rs`
- Windows build and run instructions where SQLite is a dependency

## Motivation and Context
The `tari_base_node` creates a stack overflow error when run in Windows. This has been traced to the implementation of multiple Tokio layers in `pub fn inbound_middleware_layer` (compliments due to @sdbondi). Excluding the 2x `store_forward` Tokio layers works for now. It seems this problem is related to the amount of layers added as well.

## How Has This Been Tested?
- The `tokio::runtime::Builder::new().thread_stack_size` has been increased from 2MB to 200MB but this did not solve the problem.
- Conditional compilation fully tested in Windows and Ubuntu Linux with `cargo test`

- `tari_base_node` was run and left to mine blocks on Windows and Ubuntu Linux; both mine successfully.

**Windows**
```
>> get_balance
>> Balances:
Available balance: 99716813707 µT
Pending incoming balance: 0 µT
Pending outgoing balance: 0 µT

 is not a valid command, please enter a valid command
Enter help or press tab for available commands
>> get_chain_metadata
Current meta data is is: Height of longest chain : 19
Best_block : a7cd45084c278fde1440c18784fa691d93f0de0c02b0708dda96fa308b291cb8
Pruning horizon : 2880
```

**Ubuntu Linux**
```
>> get_balance
Balances:
Available balance: 121875998510 µT
Pending incoming balance: 0 µT
Pending outgoing balance: 0 µT

>> get_chain_metadata
Current meta data is is: Height of longest chain : 22
Best_block : 73fad52710b809cf87badebd4a118183d936367fe4415d2023891fd8f5bbdc17
Pruning horizon : 2880
```

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch
* [X] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: comms/dht/src/dht.rs [hunk 1]
```diff
@@ -363,6 +367,8 @@ mod test {
         assert_eq!(msg, b"secret");
     }
 
+    // TODO: Include this test in Windows when Tokio stack overflow issue is fixed
```
Comment (sdbondi) [inline comms/dht/src/dht.rs]: nit:
```suggestion
    // FIXME: This test is excluded for Windows builds due to an unresolved stack overflow issue (#1416)
```

File: comms/dht/src/dht.rs [hunk 2]
```diff
@@ -174,7 +173,11 @@ impl Dht {
             .layer(store_forward::ForwardLayer::new(
                 Arc::clone(&self.peer_manager),
                 self.outbound_requester(),
-            ))
+            ));
+
+        // TODO: Adding these layers in Windows build create stack overflow in Tokio, to be fixed
```
Comment (sdbondi) [inline comms/dht/src/dht.rs]: nit:
```suggestion
        // FIXME: The store and forward layers are excluded for Windows builds due to an unresolved stack overflow issue (#1416)
```

Having played around with this, I don't think the issue is with tokio per se - it's more likely something to do with the complexity of the `Layer` type returned form this function and its interaction with the stack on windows. I believe one solution will be box this type (i.e move it onto the heap). An audit of what is allocated on the stack vs heap for each layer might reveal the issue.

Comment (sdbondi) [reply comms/dht/src/dht.rs]: Just a note from playing with this, it overflows regardless of which layers are added - seems to have more to do with the number of layers

File: comms/dht/src/dht.rs [hunk 3]
```diff
@@ -174,7 +173,11 @@ impl Dht {
             .layer(store_forward::ForwardLayer::new(
                 Arc::clone(&self.peer_manager),
                 self.outbound_requester(),
-            ))
+            ));
+
+        // FIXME: The store and forward layers are excluded for Windows builds due to an unresolved stack overflow issue (#1416)
```
Comment (sdbondi) [inline comms/dht/src/dht.rs]:
```suggestion
        // FIXME: The store and forward layers are excluded for Windows builds due to an unresolved stack overflow  (#1416)
```

**Comment by CjS77** [comment]
Squash the commits please

**Comment by hansieodendaal** [comment]
At least 2x more PRs will be needed after this one:
- `common\src\configuration.rs` contains Unix specific file path coding; this must be made platform agnostic. This results in Tor not working on Windows.
- To make `cargo test --workspace --all-features --jobs=3 --release` work on Windows, compilation and test of `pingpong.rs` must be prevented. This new development -  https://github.com/rust-lang/cargo/issues/2524 and https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#features - might provide a fix.
